LiteFarmOrg/LiteFarm
# Lf 2484 structures to support sensors

To test:

1. Run migrations from api folder, you should see two new db tables `sensor` and `sensor_reading`
2. Make calls to the sensor endpoints and ensure the data is stored in the db correctly.

Here is some general information to help you make the api calls
- To add a reading, the sensor that reading belongs to must exist in the db. I suggest creating sensors manually and adding readings through the api since the `add_sensor` endpoint is part of a different ticket
- Here is an example POST that ensemble may send to the `add_reading` endpoint
`[
  {
    "node_esid": "123",
    "sensor_esid": "124",
    "parameter_number": "Temperature",
    "unit": "ºC",
    "time": [
      "02-09-2022, 22:53:52",
      "02-09-2022, 22:54:07"
    ],
    "value": [
      1,
      2
    ],
    "validated": [
      false,
      false
    ]
  },
  {
    "node_esid": "123",
    "sensor_esid": "123",
    "parameter_number": "Soil Moisture",
    "unit": "kPa",
    "time": [
      "02-09-2022, 22:54:23"
    ],
    "value": [
      3
    ],
    "validated": [
      false
    ]
  }
]`
- Use '1' as the partner ID when you add a reading as this is the only partner ID we have at the moment
- When sending a request, make sure you include the appropriate JSON body
